### PR TITLE
Support COLLADA2GLTF for better imports.

### LINKED ISSUE #4885: Use COLLADA2GLTF instead of our own collada importer
### Describe the project you are working on

Godot Engine and supporting asset creators as they are.

### Describe the problem or limitation you are having in your project

Many developers are troubled by the different Collada exporters and the official Godot Engine Collada importer is not well supported.

These developers want to directly use the export from their existing Collada application with no modifications.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Use the same workflow as fbx to gltf but with collada to gltf.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

1. Duplicate the fbx2gltf code
2. Replace it with collada2gltf.
3. Submit for compatibility testing.
4. Refactor if needed for code clarity

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No.

### Is there a reason why this should be core and not an add-on in the asset library?

This is trying to move Collada import to an asset library style.

### LINKED ISSUE #51148: Dae from sketchup can not be imported properly
### Godot version

3.3.2.stable

### System information

windows10/manjaro, GLS3

### Issue description

When importing dae from Sketchup, the import gets several different errors. Importing the same dae file into blender then exporting it as dae always fixes it, but is not a working solution. Sketchup does not have GLTF support so this is the only format that can work.

Link to sketchup:
https://www.sketchup.com

### Steps to reproduce

This dae file produces this error:
https://cdn.discordapp.com/attachments/275788728168087556/871370866452226058/terrain.dae
https://cdn.discordapp.com/attachments/275788728168087556/871370304478392330/2021-08-01_14-34-20.png
editor/collada/collada.cpp:1060 - Primitive type "polygons" is not well supported (concave shapes may fail). To ensure that the geometry is properly imported, please re-export using "triangles" or "polylist".
This was fixable by changing that "polylist" setting to "triangles".

With tringles enabled, we get this:
https://cdn.discordapp.com/attachments/275788728168087556/871376248692154408/terrain.dae
image
with this error:

 editor/collada/collada.cpp:1691 - Multiple instance_* not supported.
 editor/collada/collada.cpp:1691 - Multiple instance_* not supported.
 editor/collada/collada.cpp:1691 - Multiple instance_* not supported.
 editor/collada/collada.cpp:1691 - Multiple instance_* not supported.
 editor/collada/collada.cpp:1691 - Multiple instance_* not supported.
 editor/collada/collada.cpp:1691 - Multiple instance_* not supported.
 editor/collada/collada.cpp:1691 - Multiple instance_* not supported.
 editor/collada/collada.cpp:1691 - Multiple instance_* not supported.
 editor/collada/collada.cpp:1691 - Multiple instance_* not supported.
 editor/collada/collada.cpp:1691 - Multiple instance_* not supported.
 editor/collada/collada.cpp:1691 - Multiple instance_* not supported.```
 

### Minimal reproduction project

Drop the linked dae files into a blank godot project and import.

### PR DESCRIPTION
**Pull request by fire**
Fixes: https://github.com/godotengine/godot-proposals/issues/4885
Fixes: https://github.com/godotengine/godot/issues/51148
See https://github.com/godotengine/godot-proposals/issues/2130

This also removes the Godot collada importer.

editor_screenshot_2022-07-16T055120

### CODE REVIEW

**Comment by akien-mga** [comment]
We'll need heavy testing on many types of Collada exports to see how this converter fails and make sure it's better than our ad hoc importer. But on the principle I'm all for it.

File: modules/gltf/register_types.cpp [hunk 1]
```diff
@@ -98,6 +99,26 @@ static void _editor_init() {
 			ResourceImporterScene::add_importer(importer);
 		}
 	}
+
+	// Collada importer.
```
Comment (akien-mga) [inline modules/gltf/register_types.cpp:103]:
```suggestion
	// Collada to glTF importer.
```

**Review by akien-mga — COMMENTED** [review]
Code looks great, just a few nitpicks.

**Comment by akien-mga** [comment]
You can add this too to fix capitalization in the editor settings:
```diff
diff --git a/editor/editor_property_name_processor.cpp b/editor/editor_property_name_processor.cpp
index 09d2992e07..9f4ab6f1be 100644
--- a/editor/editor_property_name_processor.cpp
+++ b/editor/editor_property_name_processor.cpp
@@ -118,6 +118,7 @@ EditorPropertyNameProcessor::EditorPropertyNameProcessor() {
 	capitalize_string_remaps["bvh"] = "BVH";
 	capitalize_string_remaps["ca"] = "CA";
 	capitalize_string_remaps["cd"] = "CD";
+	capitalize_string_remaps["collada2gltf"] = "COLLADA2GLTF";
 	capitalize_string_remaps["cpu"] = "CPU";
 	capitalize_string_remaps["csg"] = "CSG";
 	capitalize_string_remaps["db"] = "dB";
```

**Comment by akien-mga** [comment]
Tested a few scenes:

- `terrain.dae` from #51148 seems to import fine

- `torso.dae` from https://github.com/godotengine/godot/issues/14399#issuecomment-675186126 is broken. Worked OK-ish with the old importer in `master` (not sure about whether the original bug is/was still valid, skeleton seems off at least)

Screenshot_20220717_165409

- `skeleton_better.dae` and `skeleton_default.dae` from #56242 - no visible change, skeleton broken with both importers.

- `fotf_character2zy.dae` from https://github.com/godotengine/godot/issues/31005#issuecomment-750829278 - no visible change, skeleton broken with both importers. Can't say much about the bug there as the mesh doesn't move (neither with the direct .blend import)

- `Shadow.dae` from #35685 seems to import fine with the old importer (and no crash on reimport, so that bug seems fixed in current `master`). With the new importer, it's completely messed up.

image

So it's not a frank success so far, but to be fair there seems to be a lot of issues with importing animations and skeletons in general in `master` right now.

**Comment by fire** [comment]
I don't think this was a success. The work is salvageable. The importer has a self contained system that can import using anything that can take the input, output and binary flags.

In the proposal thread sketchup mentioned gltf.

**Pull request closed without merging**